core/src/main/java/org/apache/iceberg/rest/HTTPRequest.java

```diff
@@ -33,6 +33,7 @@
 @Value.Immutable
 @SuppressWarnings({"ImmutablesStyle", "SafeLoggingPropagation"})
 public interface HTTPRequest {
+  String SNAPSHOT_MODE_QUERY_PARAM_NAME = "snapshots";
```

Comment:
I don't think this is the right place either. Maybe just leave it inlined as it was before?